File: SoftUni Lessons/JavaScript Development/JS Fundamental REDO 2022/11 - Functions - Exercise/04. Odd And Even Sum .js
function oddAndEvenSum (num) {
    // MasK
    function oddSum(num) {
        return num
            .toString()
            .split('')
            .map(Number)
            .filter(n => {return n % 2 !== 0}, 0)
            .reduce((a, b) => a + b, 0);
    }

    function evenSum(num) {
        return num
            .toString()
            .split('')
            .map(Number)
            .filter(n => {return n % 2 === 0}, 0)
            .reduce((a, b) => a + b, 0);
    }

    console.log(`Odd sum = ${oddSum(num)}, Even sum = ${evenSum(num)}`);
}

oddAndEvenSum(1000435);  //  Odd sum = 9, Even sum = 4

File: SoftUni Lessons/Python Development/Python Fundamentals May 2023/11. Functions - Exercise/04. Odd and Even Sum.py
def odd_and_even_sum(num):
    odd_sum = sum(list(filter(lambda x: x % 2 != 0, list(map(int, list(str(num)))))))
    even_sum = sum(list(filter(lambda x: x % 2 == 0, list(map(int, list(str(num)))))))

    return f"Odd sum = {odd_sum}, Even sum = {even_sum}"


print(odd_and_even_sum(int(input())))
